File: rb/features.rb
# SlaUptimeCalculator SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module SlaUptimeCalculatorFeatures
  def self.make_feature(name)
    case name
    when "base"
      SlaUptimeCalculatorBaseFeature.new
    when "ratelimit"
      SlaUptimeCalculatorRatelimitFeature.new
    when "retry"
      SlaUptimeCalculatorRetryFeature.new
    when "test"
      SlaUptimeCalculatorTestFeature.new
    when "timeout"
      SlaUptimeCalculatorTimeoutFeature.new
    else
      SlaUptimeCalculatorBaseFeature.new
    end
  end
end

File: py/features.py
# SlaUptimeCalculator SDK feature factory

from feature.base_feature import SlaUptimeCalculatorBaseFeature
from feature.test_feature import SlaUptimeCalculatorTestFeature


def _make_feature(name):
    features = {
        "base": lambda: SlaUptimeCalculatorBaseFeature(),
        "test": lambda: SlaUptimeCalculatorTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
